Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -355,6 +355,125 @@ void RewriteGlobalVariables::visitEntryPoint(AST::Function& function, AST::Stage
     insertLocalDefinitions(function, usedGlobals.privateGlobals);
 }
 
+static BindGroupLayoutEntry::BindingMember bindingMemberForGlobal(auto& global)
+{
+    auto* variable = global.declaration;
+    ASSERT(variable);
+    auto* type = variable->storeType();
+    ASSERT(type);
+
+    using namespace WGSL::Types;
+    return WTF::switchOn(*type, [&](const Primitive& primitive) -> BindGroupLayoutEntry::BindingMember {
+        switch (primitive.kind) {
+        case Types::Primitive::AbstractInt:
+        case Types::Primitive::I32:
+        case Types::Primitive::U32:
+        case Types::Primitive::AbstractFloat:
+        case Types::Primitive::F32:
+        case Types::Primitive::Void:
+        case Types::Primitive::Bool:
+            RELEASE_ASSERT_NOT_REACHED();
+        case Types::Primitive::Sampler:
+            return SamplerBindingLayout {
+                .type = SamplerBindingType::Filtering
+            };
+        case Types::Primitive::TextureExternal:
+            return ExternalTextureBindingLayout { };
```

Comment:
maybe add a `FIXME`?

Reply:
No `FIXME` needed, this is fully implemented. The struct has no members.